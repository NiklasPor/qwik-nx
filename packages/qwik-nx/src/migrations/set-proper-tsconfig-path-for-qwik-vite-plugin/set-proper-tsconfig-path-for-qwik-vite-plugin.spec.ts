import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';

import update from './set-proper-tsconfig-path-for-qwik-vite-plugin';
import appGenerator from '../../generators/application/generator';

describe('set-proper-tsconfig-path-for-qwik-vite-plugin migration', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await appGenerator(tree, { name: 'myapp' });
  });

  it('should add "tsconfigFileNames" property', async () => {
    const viteConfigPath = `apps/myapp/vite.config.ts`;
    // just to ensure the file at that paths exists before updating it
    expect(tree.exists(viteConfigPath)).toBeTruthy();

    tree.write(viteConfigPath, getViteConfigSample());
    update(tree);
    expect(tree.read(viteConfigPath, 'utf-8')).toMatchSnapshot();
  });
  it('should not modify the "tsconfigFileNames" property if it exists', async () => {
    const viteConfigPath = `apps/myapp/vite.config.ts`;
    tree.write(viteConfigPath, getViteConfigSample(true));

    update(tree);

    expect(tree.read(viteConfigPath, 'utf-8')).toMatchSnapshot();
  });
});

const getViteConfigSample = (
  includeTSConfigFileNamesProp = false
) => `import { qwikVite } from '@builder.io/qwik/optimizer';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  plugins: [
    qwikNxVite(),
    qwikVite({
      client: {
        outDir: '../../dist/apps/myapp/client',
      },
      ssr: {
        outDir: '../../dist/apps/myapp/server',
      },${
        includeTSConfigFileNamesProp
          ? '\ntsconfigFileNames: ["MOCK_VALUE"]'
          : ''
      }
    }),
    tsconfigPaths({ root: '../../' }),
  ],
});
`;
