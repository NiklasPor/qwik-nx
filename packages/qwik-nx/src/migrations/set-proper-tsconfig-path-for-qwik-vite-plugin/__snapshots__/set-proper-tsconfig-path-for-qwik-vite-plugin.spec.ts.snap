// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set-proper-tsconfig-path-for-qwik-vite-plugin migration should add "tsconfigFileNames" property 1`] = `
"import { qwikVite } from \\"@builder.io/qwik/optimizer\\";
import { defineConfig } from \\"vite\\";
import tsconfigPaths from \\"vite-tsconfig-paths\\";
import { qwikNxVite } from \\"qwik-nx/plugins\\";
export default defineConfig({
    plugins: [
        qwikNxVite(),
        qwikVite({
            tsconfigFileNames: [\\"tsconfig.app.json\\"],
            client: {
                outDir: \\"../../dist/apps/myapp/client\\",
            },
            ssr: {
                outDir: \\"../../dist/apps/myapp/server\\",
            }
        }),
        tsconfigPaths({ root: \\"../../\\" })
    ]
});
"
`;

exports[`set-proper-tsconfig-path-for-qwik-vite-plugin migration should not modify the "tsconfigFileNames" property if it exists 1`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  plugins: [
    qwikNxVite(),
    qwikVite({
      client: {
        outDir: '../../dist/apps/myapp/client',
      },
      ssr: {
        outDir: '../../dist/apps/myapp/server',
      },
tsconfigFileNames: [\\"MOCK_VALUE\\"]
    }),
    tsconfigPaths({ root: '../../' }),
  ],
});
"
`;
